lua-api/lib/src/ScriptContext.cpp

```diff
@@ -0,0 +1,1011 @@
+    }
+    case L"FloatProperty"_fnv:
+        return sol::make_object(s, *(float*)((uintptr_t)self + offset));
+    case L"DoubleProperty"_fnv:
+        return sol::make_object(s, *(double*)((uintptr_t)self + offset));
+    case L"IntProperty"_fnv:
+        return sol::make_object(s, *(int32_t*)((uintptr_t)self + offset));
+    case L"UIntProperty"_fnv:
+    case L"UInt32Property"_fnv:
+        return sol::make_object(s, *(uint32_t*)((uintptr_t)self + offset));
+    case L"NameProperty"_fnv:
+        return sol::make_object(s, *(uevr::API::FName*)((uintptr_t)self + offset));
+    case L"ObjectProperty"_fnv:
+        return sol::make_object(s, *(uevr::API::UObject**)((uintptr_t)self + offset));
+    case L"ArrayProperty"_fnv:
+    {
+        const auto inner_prop = ((uevr::API::FArrayProperty*)desc)->get_inner();
+
+        if (inner_prop == nullptr) {
+            return sol::make_object(s, sol::lua_nil);
+        }
+
+        const auto inner_c = inner_prop->get_class();
+
+        if (inner_c == nullptr) {
+            return sol::make_object(s, sol::lua_nil);
+        }
+
+        const auto inner_name_hash = utility::hash(inner_c->get_fname()->to_string());
+
+        switch (inner_name_hash) {
+        case "ObjectProperty"_fnv:
+        {
+            const auto& arr = *(uevr::API::TArray<uevr::API::UObject*>*)((uintptr_t)self + offset);
+
+            if (arr.data == nullptr || arr.count == 0) {
+                return sol::make_object(s, sol::lua_nil);
+            }
+
+            auto lua_arr = std::vector<uevr::API::UObject*>{};
+
+            for (size_t i = 0; i < arr.count; ++i) {
+                lua_arr.push_back(arr.data[i]);
+            }
+
+            return sol::make_object(s, lua_arr);
+        }
+        // TODO: Add support for other types
+        };
+
+        return sol::make_object(s, sol::lua_nil);
+    }
+    };
+
+    return sol::make_object(s, sol::lua_nil);
+}
+
+sol::object prop_to_object(sol::this_state s, uevr::API::UObject* self, const std::wstring& name) {
+    const auto c = self->get_class();
+
+    if (c == nullptr) {
+        return sol::make_object(s, sol::lua_nil);
+    }
+
+    const auto desc = c->find_property(name.c_str());
+
+    if (desc == nullptr) {
+        if (auto fn = c->find_function(name.c_str()); fn != nullptr) {
+            /*return sol::make_object(s, [self, s, fn](sol::variadic_args args) {
+                return call_function(s, self, fn, args);
+            });*/
+            return sol::make_object(s, fn);
+        }
+
+        return sol::make_object(s, sol::lua_nil);
+    }
+
+    return prop_to_object(s, self, desc);
+}
+
+void set_property(sol::this_state s, uevr::API::UObject* self, const std::wstring& name, sol::object value) {
+    const auto c = self->get_class();
+
+    if (c == nullptr) {
+        return;
+    }
+
+    const auto desc = c->find_property(name.c_str());
+
+    if (desc == nullptr) {
+        return;
```

Comment:
should these early returns log something to help the script developer? If there property / class isn't found (typo, wrong path, whatever) then the failing set_property call might go unnoticed

Reply:
yeah, I can add a throw